doc/explanation/ubuntu-image-security-overview.rst

```diff
@@ -0,0 +1,55 @@
+.. _ubuntu-image-security-overview:
+
+ubuntu-image security overview
+==============================
+
+Overview of security aspects of ubuntu-image.
+
+
+Privileged execution
+--------------------
+ 
+``ubuntu-image`` requires **elevated permissions** to properly run. It is recommended to use a dedicated building machine, make sure ``ubuntu-image`` is installed from a trusted source and the provided configuration is trusted. 
+
+
+Cryptography
+------------
+
+``ubuntu-image`` is a wrapper around several lower level tools to build a Ubuntu image. Use of cryptographic technologies can then be grouped in two categories: direct usage in the tool or in one of the wrapped tools.
+
+
+ubuntu-image itself and Go libraries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Tarball SHA256 checksum: When building a ``classic`` image, user can provide a tarball serving as the rootfs content of the image. The SHA256 checksum of the tarball can optionally be provided to verify it has not been altered. This verification `crypto/sha256`_ Go standard library package, implementing the SHA256 hash algorithm as defined in FIPS 180-4.
+
+* Disk UUIDs: To generate unique disk IDs, ``ubuntu-image`` uses the ``Read()`` function from the `crypto/rand`_ Go standard library package, implementing a cryptographically secure random number generator.
+
+* Model assertion signature: When building a ``snap`` image, user must provide a signed model assertion. Verification of this signature is handled by the `snapd library`_, relying on SHA3-384 and SHA512 for hashing and OpenPGP v4 signatures with RSA 4096/8192 keys.
+
+
+Underlying tools
+~~~~~~~~~~~~~~~~
+
+* PPA handling: During the build, PPAs can be added to the ``apt`` configuration of the image. The fingerprint of the PPA can be declared in the configuration provided to ``ubuntu-image``. This value is passed down to ``gpg`` to fetch the keys to validate the packages signature. The correct handling of these keys is delegated to ``gpg`` up to the point where ``ubuntu-image`` write them to the resulting ``apt`` configuration.
```

Comment:
```suggestion
* PPA handling: During the build, PPAs (Personal Package Archives) can be added to the ``apt`` configuration of the image. The fingerprint of the PPA can be declared in the configuration provided to ``ubuntu-image``. This value is passed to ``gpg`` to fetch the keys to validate the package signatures. The correct handling of these keys is delegated to ``gpg`` up to the point where ``ubuntu-image`` writes them to the resulting ``apt`` configuration.
```

Reply:
I am not sure about the suggestion to replace `packages signature` with `package signature`. I was talking about the signature of each package, not the multiple signatures of a single package.